src/beamlime/applications/_nexus_helpers.py

```diff
@@ -2,11 +2,56 @@
 # Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
 import os
 from functools import partial
-from typing import Any, Callable, Optional, Union
+from types import MappingProxyType
+from typing import Any, Callable, Literal, NamedTuple, Optional, Union
+
+import numpy as np
 
 Path = Union[str, bytes, os.PathLike]
 
 
+def _nested_mapping_proxy(obj: dict) -> MappingProxyType:
+    """Create a nested shallow copy/mapping proxy of ``obj``.
+
+    If any value in the dictionary is a dictionary,
+    it is converted to a mapping proxy type.
+
+    Examples
+    --------
+    >>> from types import MappingProxyType
+    >>> _nested_mapping_proxy_type({'a': {'b': 1}})
+    mappingproxy({'a': mappingproxy({'b': 1})})
+
+    """
+    return MappingProxyType(
+        {
+            k: _nested_mapping_proxy(v) if isinstance(v, dict) else v
+            for k, v in obj.items()
+        }
+    )
+
+
+class DatasetRecipe(NamedTuple):
+    dtype: type
+    ndim: int
+
+
+FBIdentifier = Literal["ev44", "f144", "tdct"]
+ValuesRecipe = MappingProxyType[str, DatasetRecipe]
+GROUP_RECIPES: MappingProxyType[FBIdentifier, ValuesRecipe] = _nested_mapping_proxy(
```

Comment:
Why do we need the `_nested_mapping_proxy`?

Reply:
I wanted to keep it read-only... but since the ``DatasetRecipe`` is a data class, it won't be read-only anyways. I'll just remove it.